mail_outbound_static/views/ir_mail_server_view.xml

```diff
@@ -14,6 +14,7 @@
         <field name="arch" type="xml">
             <xpath expr="//field[@name='smtp_pass']" position="after">
                 <field name="smtp_from" />
+                <field name="domain_whitelist" attrs="{'invisible': [('smtp_from', '=', False)]}"/>
```

Comment:
may be we could keep always visible, because actually the functionality to select a proper mailserver is still there if you just set this field (set smtp_from, set domain_whitelist, remove smtp_from)